File: client-domain/src/test/java/com/jraska/github/client/time/TestTimeProvider.java
package com.jraska.github.client.time;

public final class TestTimeProvider implements TimeProvider {
  private long elapsed;

  public TestTimeProvider() {
    this(0);
  }

  public TestTimeProvider(long elapsed) {
    this.elapsed = elapsed;
  }

  @Override public long elapsed() {
    return elapsed;
  }

  public TestTimeProvider elapsed(long elapsedMillis) {
    this.elapsed = elapsedMillis;
    return this;
  }

  public TestTimeProvider advanceTime(long difference) {
    if (difference < 0) {
      throw new IllegalArgumentException("Can only advance time forward");
    }

    elapsed += difference;
    return this;
  }
}

File: core/src/test/java/com/jraska/github/client/time/TestTimeProvider.kt
package com.jraska.github.client.time

class TestTimeProvider @JvmOverloads constructor(private var elapsed: Long = 0) : TimeProvider {

  override fun elapsed(): Long {
    return elapsed
  }

  fun elapsed(elapsedMillis: Long): TestTimeProvider {
    this.elapsed = elapsedMillis
    return this
  }

  fun advanceTime(difference: Long): TestTimeProvider {
    if (difference < 0) {
      throw IllegalArgumentException("Can only advance time forward")
    }

    elapsed += difference
    return this
  }
}
